client/question/questionSpec.js

```diff
@@ -95,8 +95,18 @@ describe('question.js', function() {
       RUNTIME_ERROR_FEEDBACK_MESSAGES.python.forEach(function(error) {
         expect(error.checker('test')).toBe(false);
         expect(typeof error.generateMessage(['NameError: name \'key\' is not ',
-          'defined KeyError: key AttributeError: \'key\' object has no ',
-          'attribute \'length\''].join(''))).toEqual('string');
+          'defined KeyError: key blah on line 47 AttributeError: \'key\' ',
+          'object has no attribute \'length\''].join(''))).toEqual('string');
+      });
+    });
+
+    it('should handle an empty key', function() {
+      var errorString = 'KeyError: key "" on line 47';
+      RUNTIME_ERROR_FEEDBACK_MESSAGES.python.forEach(function(error) {
+        if (error.checker(errorString)) {
+          expect(typeof error.generateMessage(
```

Comment:
This should be: expect(error.generateMessage(...)).toEqual(...) since the point is to test the generated string.